File: 0258-add-digits/0258-add-digits.rb
# @param {Integer} num
# @return {Integer}
def reduce(n)
  ans = 0
  while n > 0
    ans += n%10
    n /= 10
  end
  ans
end

def add_digits(num)
  num = reduce(num) while num >= 10
  num
end

File: 0258-add-digits/0258-add-digits.ts
function addDigits(num: number): number {
  function reduce(n: number): number {
    let ans = 0
    while (n > 0) {
      ans += n % 10
      n = Math.floor(n / 10)
    }
    return ans
  }
  
  while (num >= 10) num = reduce(num)
  return num
}